File: eco-core/core-plugin/src/main/java/com/willfp/reforges/commands/CommandReforge.java
package com.willfp.reforges.commands;


import com.willfp.eco.core.EcoPlugin;
import com.willfp.eco.core.command.CommandHandler;
import com.willfp.eco.core.command.impl.PluginCommand;
import com.willfp.reforges.gui.ReforgeGUI;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

public class CommandReforge extends PluginCommand {
    /**
     * Instantiate a new command handler.
     *
     * @param plugin The plugin for the commands to listen for.
     */
    public CommandReforge(@NotNull final EcoPlugin plugin) {
        super(plugin, "reforge", "reforges.command.reforge", true);
    }

    @Override
    public CommandHandler getHandler() {
        return (sender, args) -> {
            ReforgeGUI.getMenu().open((Player) sender);
        };
    }
}

File: eco-core/core-plugin/src/main/kotlin/com/willfp/reforges/commands/CommandReforge.kt
package com.willfp.reforges.commands

import com.willfp.eco.core.command.impl.PluginCommand
import com.willfp.eco.core.sound.PlayableSound
import com.willfp.reforges.gui.ReforgeGUI
import com.willfp.reforges.plugin
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

object CommandReforge : PluginCommand(
    plugin,
    "reforge",
    "reforges.command.reforge",
    true
) {
    @Suppress("DEPRECATION")
    override fun onExecute(player: CommandSender, args: List<String>) {
        player as Player
        PlayableSound.create(plugin.configYml.getSubsection("gui.open-sound"))?.playTo(player)
        ReforgeGUI.open(player)
    }
}
